bigquery-connector-common/src/main/java/com/google/cloud/bigquery/connector/common/BigQueryUtil.java

```diff
@@ -570,15 +575,17 @@ public static Schema adjustSchemaIfNeeded(Schema wantedSchema, Schema existingTa
    * @return the adjusted field
    */
   @VisibleForTesting
-  static Field adjustField(Field field, @Nullable Field existingField) {
-    if (field.getType().equals(LegacySQLTypeName.NUMERIC)
-        && existingField != null
-        && existingField.getType().equals(LegacySQLTypeName.BIGNUMERIC)) {
-      // convert type
-      return field.toBuilder().setType(LegacySQLTypeName.BIGNUMERIC).build();
+  static Field adjustField(
+      Field field, @Nullable Field existingField, boolean allowFieldRelaxation) {
+    if (existingField == null) {
+      return field;
     }
+    Mode fieldMode = existingField.getMode() != null ? existingField.getMode() : NULLABLE;
```

Comment:
condition is redundant as you know that `existingField.getMode() != null` based on line 580